### DISCUSSION TITLE
entity.combine() not working

### DISCUSSION THREAD

**Discussion by JacktheFabricRipper** [pokepetter/ursina#307] [Q&A]
Whenever I use combine(), the entities do not take on the properties of the parent. They all get no collider, no texture, black color, no shader, even if I change it. I have some code below, is there anything I am doing wrong?

from ursina import *
from ursina.shaders import basic_lighting_shader
from random import *
from perlin_noise import PerlinNoise

app = Ursina()

terrain = Entity(model=None, collider=None)
noise = PerlinNoise(octaves=2, seed=seed())

for x in range(20):
    for z in range(20):
        y = round(noise([x/10,z/10])*5)
        block = Entity(model='cube')
        block.x = x
        block.y = y
        block.z = z
        block.parent = terrain

terrain.combine()
terrain.texture = 'white_cube'
terrain.color = color.white
terrain.shader = basic_lighting_shader

EditorCamera()

app.run()

**Comment by pokepetter** [comment]
Looks like there's a rogue `continue` in the combine code :P
That's for the bug report, will fix as soon as possible.

**Reply by JacktheFabricRipper** [reply]
Did you update it? I updated it in pip but it says it doesn't have a new version...

**Reply by pokepetter** [reply]
Yes, get it from GitHub
```pip install https://github.com/pokepetter/ursina/archive/master.zip --upgrade```

**Reply by JacktheFabricRipper** [reply]
I just tried again and it didn't work. It still was black.

**Reply by pokepetter** [reply]
Try uninstalling and reinstalling. The code you posted works correctly now, for me at least.

**Reply by JacktheFabricRipper** [reply]
k
